pkg/modules/spanpercentile/translator.go

```diff
@@ -0,0 +1,116 @@
+package spanpercentile
+
+import (
+	"fmt"
+	"github.com/SigNoz/signoz/pkg/querybuilder"
+	"github.com/SigNoz/signoz/pkg/telemetrytraces"
+	qbtypes "github.com/SigNoz/signoz/pkg/types/querybuildertypes/querybuildertypesv5"
+	"github.com/SigNoz/signoz/pkg/types/spanpercentiletypes"
+	"github.com/huandu/go-sqlbuilder"
+)
+
+func BuildSpanPercentileQuery(req *spanpercentiletypes.SpanPercentileRequest) (*qbtypes.QueryRangeRequest, error) {
+	if err := req.Validate(); err != nil {
+		return nil, err
+	}
+
+	startNano := querybuilder.ToNanoSecs(req.Start)
+	endNano := querybuilder.ToNanoSecs(req.End)
+	startBucketSec := req.Start/1000 - querybuilder.BucketAdjustment
+	endBucketSec := req.End / 1000
+
+	targetSpanBuilder := sqlbuilder.NewSelectBuilder()
+	targetSpanBuilder.Select(
+		"duration_nano",
+		"name",
+		"resource_string_service$$name as `service_name`",
+		"resources_string['deployment.environment'] as deployment_environment",
```

Comment:
Whenever you are using resource based filtering, please add why fingerprint is not used if we don't use it.